streaming/src/main/scala/org/apache/spark/streaming/rdd/WriteAheadLogBackedBlockRDD.scala

```diff
@@ -108,9 +110,13 @@ class WriteAheadLogBackedBlockRDD[T: ClassTag](
           // writing log data. However, the directory is not needed if data needs to be read, hence
           // a dummy path is provided to satisfy the method parameter requirements.
           // FileBasedWriteAheadLog will not create any file or directory at that path.
-          val dummyDirectory = FileUtils.getTempDirectoryPath()
+          // FileBasedWriteAheadLog will not create any file or directory at that path. Also,
+          // this dummy directory should not already exist otherwise the WAL will try to recover
+          // past events from the directory and throw errors.
+          val nonExistentDirectory = new File(
+            FileUtils.getTempDirectory(), UUID.randomUUID().toString).getAbsolutePath
```

Comment:
I keep trying to get rid of uses of commons-io since it is not actually a Spark dependency. This can be `Utils.createTempDir`? The other usage in `StreamingContextSuite` can be replaced with Guava. Not exactly the topic of this PR but might yet be worth doing if you're potentially removing this usage.